RealmSwift/Realm.swift

```diff
@@ -1057,17 +1054,20 @@ extension Realm {
                 fallthrough
             }
         case .always:
-            try await withCheckedThrowingContinuation { (continuation: CheckedContinuation<Void, Swift.Error>) in
-                RLMRealm.asyncOpen(with: configuration.rlmConfiguration, callback: { error in
+            rlmRealm = try await withCheckedThrowingContinuation { continuation in
+                RLMRealm.asyncOpen(with: configuration.rlmConfiguration, callbackQueue: .main) { (realm, error) in
```

Comment:
Is specifying the callback queue needed here?

Reply:
That is what makes it so that we can use the RLMRealm instance from the asyncOpen rather than obtaining a new one.

Reply:
Right, I thought `@MainActor` would've covered this for us, but fair enough.